src/App.test.js

```diff
@@ -16,9 +16,17 @@ it("renders without crashing", () => {
   shallow(<App />);
 });
 
-it("snapshot testing that it renders correctly", () => {
+it("renders correctly (snapshot testing)", () => {
   const wrapper = mount(<App />);
   expect(wrapper).toMatchSnapshot();
+  wrapper.unmount();
+});
+
+it("renders LogoutButton when authenticationInfo state changes", () => {
+  const wrapper = mount(<App />);
+  wrapper.setState({authenticationInfo: {access_token: "some_access_token", expires_in: "30"}});
```

Comment:
I don't think this spec is correct - "renders LogoutButton when authenticationInfo state changes". We don't want to show Logout if the authInfo changed from a valid object to undefined, for example. 

It should say something like "renders LogoutButton when state changes to contain a valid authInfo."

Then, I think we should make the test more explicit. How about assign state to undefined first and then set a valid auth info?

Reply:
On second thoughts, aren't you testing beyond the boundaries of your 'unit'? Shouldn't you just test that isLogged value is propagated to a 'shallow' component? You can always change the state to contain different authInfo objects to check that the right isLogged value is generated.